tests/test_core.py

```diff
@@ -94,6 +97,57 @@ def test_get_executor_args(package_path, expected_path):
             assert executor == expected_executor
     else:
         core.normalize(package_path, dry_run=True)
+
+
+@pytest.mark.parametrize(
+    'package_path, build_args_envs',
+    [   
+        (
+            Path(__file__).parent / 'cases' / 'executor_1',
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            }
+        ),
+        (
+            Path(__file__).parent / 'cases' / 'executor_7',
+            { 
+                'AUTH_TOKEN': "AUTH_TOKEN",
+                'TOKEN': 'ghp_Nwh9o70GDSzs'
+            }
+        ),
+    ],
+)
+def test_compare_dockerfile_env_vars(package_path, build_args_envs):
+
+    dockerfile_path = Path(package_path / 'Dockerfile') 
+    dockerfile_expected_path = Path(package_path / 'Dockerfile.expect') 
+
+    originDockerfileStr = None;
+    if dockerfile_path.exists():
+        with open(dockerfile_path, 'r') as fp:
+            originDockerfileStr = str(fp.read())
+
+    core.normalize(package_path, build_args_envs=build_args_envs, dry_run=False)
+    if dockerfile_path.exists():
+        dockerfileStr = None
+        with open(dockerfile_path, 'r') as fp:
+            dockerfileStr = str(fp.read())
+        
+        dockerfileExpectedStr = ''
+        with open(dockerfile_expected_path, 'r') as fp:
+            dockerfileExpectedStr = str(fp.read())
+
+        assert dockerfileExpectedStr == dockerfileStr
+
+
+    if originDockerfileStr:
+        with open(dockerfile_path, 'w') as fp:
+            fp.write(originDockerfileStr)
+    elif dockerfile_path.exists():
+        os.remove(dockerfile_path)
```

Comment:
If the `assert` failed, then the clear logic won't be executed.

Reply:
yes, fix it